README.md

```diff
@@ -121,6 +122,91 @@ Select a provider from the following list, then follow the installation steps fo
+```yaml
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: my-provider
+spec:
+  provider: vault                             # accepted provider options: azure or vault
+  parameters:                                 # provider-specific parameters
+```
+
+Here is a sample [`SecretProviderClass` custom resource](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/test/bats/tests/vault_v1alpha1_secretproviderclass.yaml)
+
+### Update your Deployment Yaml
+
+To ensure your application is using the Secrets Store CSI driver, update your deployment yaml to use the `secrets-store.csi.k8s.io` driver and reference the `SecretProviderClass` resource created in the previous step.
+
+```yaml
+volumes:
+  - name: secrets-store-inline
+    csi:
+      driver: secrets-store.csi.k8s.io
+      readOnly: true
+      volumeAttributes:
+        secretProviderClass: "my-provider"
+```
+
+Here is a sample [deployment yaml](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/test/bats/tests/nginx-pod-vault-inline-volume-secretproviderclass.yaml) using the Secrets Store CSI driver.
+
+### Secret Content is Mounted on Pod Start
+On pod start and restart, the driver will call the provider binary to retrieve the secret content from the external Secrets Store you have specified in the `SecretProviderClass` custom resource. Then the content will be mounted to the container's file system. 
+
+To validate, once the pod is started, you should see the new mounted content at the volume path specified in your deployment yaml.
+
+```bash
+kubectl exec -it nginx-secrets-store-inline ls /mnt/secrets-store/
+foo
+```
+
+### [OPTIONAL] Sync with Kubernetes Secrets
+
+In some cases, you may want to create a Kubernetes Secret to mirror the mounted content. Use the optional `secretObjects` field to define the desired state of the synced Kubernetes secret objects.
+
+A `SecretProviderClass` custom resource should have the following components:
+```yaml
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: my-provider
+spec:
+  provider: vault                             # accepted provider options: azure or vault
+  parameters:                                 # provider-specific parameters
+  secretObjects:                              # [OPTIONAL] SecretObject defines the desired state of synced K8s secret objects
+  - data:
+    - key: username                           # data field to populate
+      objectName: foo1                        # name of the object to sync
+    secretName: foosecret                     # name of the Kubernetes Secret object
+    type: Opaque                              # type of the Kubernetes Secret object e.g. Opaque, kubernetes.io/tls
+```
+> NOTE: Here is the list of [supported Kubernetes Secret types](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/pkg/secrets-store/utils.go#L660-L675). 
```

Comment:
We should probably add the list here. The line #s will change as we make changes to the code.

Reply:
Not a blocker for this PR. We'll update this the next time.